dom.d.ts

```diff
@@ -0,0 +1,44 @@
+type Nil = null | undefined;
+type Primitive = string | number | bigint | boolean;
+
+type Prop = Nil | Primitive;
+type Props = Nil | Record<string, Prop>;
+
+type Child = Nil | Primitive | Node | Child[];
+
+export function E<N extends HTMLInputElement>(name: 'input', props?: Props, ...children: Child[]): N;
+export function E<N extends HTMLElement>(name: string, props?: Props, ...children: Child[]): N;
+
+export function S(name: string, props?: Props, ...children: Child[]): SVGElement;
+
+export function F(...children: Child[]): Node;
+
+export function reset<N extends Node>(node: N | Nil, props?: Props, ...children: Child[]): N | Nil;
+
+export function resetProps<N extends Node>(node: N | Nil, props?: Props): N | Nil;
+
+export function replace<N extends Node>(node: N | Nil, ...children: Child[]): void;
+
+type ClsVal = Nil | Primitive | Array<ClsVal>;
```

Comment:
Technically, this throws on non-falsy non-string primitives such as `1`. This seems hard to express in the type signature. Might change the behavior instead. Hmm...